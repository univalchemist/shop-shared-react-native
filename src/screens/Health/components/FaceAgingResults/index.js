import React, { useState } from 'react';
import { View, StyleSheet } from 'react-native';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { Slider } from 'react-native-elements';
import {
  Box,
  Flex,
  Image,
  Loader,
  ScreenHeadingText,
  Text,
} from '@wrappers/components';
import theme from '@theme';
import PropTypes from 'prop-types';
import { useIntl } from '@wrappers/core/hooks';
import { ErrorCard } from '@screens/Health/components/widgets/ErrorCard.js';
import { CardContainer } from '@screens/Health/components/widgets/CardContainer';
import { categories, logAction } from '@store/analytics/trackingActions';

const styles = StyleSheet.create({
  container: { paddingHorizontal: 32, paddingBottom: 32 },
});

export const FaceAgingResults = ({ faceAging }) => {
  const intl = useIntl();
  const [selectedAgeIndex, setSelectedAgeIndex] = useState(0);
  return (
    <View style={styles.container}>
      {faceAging.isError ? (
        renderFaceAgingErrorPanel({ intl })
      ) : faceAging.faceAgingIsDone &&
        faceAging.expectedTotalResults === faceAging.currentTotalResults ? (
        renderFaceAgingResults(
          faceAging,
          selectedAgeIndex,
          setSelectedAgeIndex,
          intl,
        )
      ) : (
        <>
          <ScreenHeadingText mb={2}>
            {intl.formatMessage({ id: 'futureMeTitle' })}
          </ScreenHeadingText>
          <CardContainer
            theme={theme}
            justifyContent="center"
            color={theme.colors.white}
            flexGrow={1}
          >
            <Loader
              primary
              textAlign={'center'}
              loadingText={intl.formatMessage({
                id: 'health.healthScoreHistory.lifestyleImage.processing',
                defaultMessage:
                  'Analysing in progress. This may take a little while. Please do not navigate away.',
              })}
            />
          </CardContainer>
        </>
      )}
    </View>
  );
};

const renderFaceAgingResults = (
  faceAging,
  selectedAgeIndex,
  setSelectedAgeIndex,
  intl,
) => {
  let faceAges = [];
  if (faceAges.length === 0) {
    faceAges = Object.keys(faceAging.results).sort();
  }
  const selectedAge = faceAges[selectedAgeIndex];
  const selectedAgeForDisplay = selectedAge == '67' ? '65' : selectedAge;

  return faceAges.length > 0 ? (
    <>
      <ScreenHeadingText mb={2}>
        {intl.formatMessage({ id: 'futureMeTitle' })}
      </ScreenHeadingText>
      <Box>
        <Text
          accessible={true}
          accessibilityLabel={intl.formatMessage(
            {
              id: 'health.sectionTitle.faceAging',
            },
            { selectedAge: selectedAgeForDisplay },
          )}
          color={theme.colors.gray[0]}
          fontWeight={theme.fontWeights.bold}
          fontSize={16}
          lineHeight={22}
          letterSpacing={0.3}
          mt={8}
          mb={8}
        >
          {intl.formatMessage(
            {
              id: 'health.sectionTitle.faceAging',
            },
            { selectedAge: selectedAgeForDisplay },
          )}
        </Text>
        {renderFaceAgingImages(selectedAge, faceAging, intl)}
        <Box mt={16}>
          <Slider
            style={{ width: '100%', height: 40 }}
            step={1}
            minimumValue={0}
            maximumValue={faceAges.length - 1}
            thumbTintColor={theme.colors.black}
            minimumTrackTintColor={theme.colors.black}
            maximumTrackTintColor={theme.colors.gray[4]}
            onValueChange={faceAgeIndex => {
              setSelectedAgeIndex(faceAgeIndex);
            }}
            onSlidingComplete={() => {
              logAction({
                category: categories.LIFESTYLE_OVERVIEW,
                action: `Slider age ${selectedAgeForDisplay}`,
              });
            }}
          />
          <Flex flex={1} flexDirection="row" />
        </Box>
        <Text
          accessibilityLabel={intl.formatMessage({
            id: 'health.faceAgingDescription',
          })}
          accessible={true}
          fontSize={14}
          mt={8}
          mb={16}
        >
          {intl.formatMessage({ id: 'health.faceAgingDescription' })}
        </Text>
      </Box>
    </>
  ) : (
    <Box />
  );
};

const renderFaceAgingImages = (selectedAge, faceAging, intl) => {
  return (
    <>
      {Object.keys(faceAging.results).map((faceAge, index) => {
        return (
          <Flex
            key={index}
            flexDirection="row"
            style={faceAge === selectedAge ? {} : { display: 'none' }}
          >
            <Box flex={1} marginRight={8}>
              <Box>
                <Image
                  accessibilityLabel={intl.formatMessage({
                    id:
                      'health.accessibilityLabel.LivingHealthyLifestyleFaceAgingImage',
                  })}
                  accessible={true}
                  accessibilityRole={'image'}
                  height={210}
                  source={{ uri: faceAging.results[faceAge].healthy }}
                  borderRadius={5}
                />
                <Text
                  accessibilityLabel={intl.formatMessage({
                    id: 'health.LivingHealthyLifestyle',
                  })}
                  accessible={true}
                  fontSize={14}
                  paddingTop={8}
                >
                  {intl.formatMessage({ id: 'health.LivingHealthyLifestyle' })}
                </Text>
              </Box>
            </Box>
            <Box flex={1}>
              <Box>
                <Image
                  accessibilityLabel={intl.formatMessage({
                    id:
                      'health.accessibilityLabel.LivingCurrentLifestyleFaceAgingImage',
                  })}
                  accessible={true}
                  accessibilityRole={'image'}
                  height={210}
                  source={{ uri: faceAging.results[faceAge].unhealthy }}
                  borderRadius={5}
                />
                <Text
                  accessibilityLabel={intl.formatMessage({
                    id: 'health.LivingCurrentLifestyle',
                  })}
                  accessible={true}
                  fontSize={14}
                  paddingTop={8}
                >
                  {intl.formatMessage({ id: 'health.LivingCurrentLifestyle' })}
                </Text>
              </Box>
            </Box>
          </Flex>
        );
      })}
    </>
  );
};

const renderFaceAgingErrorPanel = ({ intl }) => (
  <Box>
    <ScreenHeadingText mb={2}>
      {intl.formatMessage({ id: 'futureMeTitle' })}
    </ScreenHeadingText>
    <ErrorCard color={theme.colors.white} />
  </Box>
);

const mapStateToProps = ({ health: { faceAging } }) => ({
  faceAging,
});

const enhance = compose(connect(mapStateToProps));

export default enhance(FaceAgingResults);

FaceAgingResults.propTypes = {
  faceAging: PropTypes.shape({
    faceAgingIsDone: PropTypes.bool,
    expectedTotalResults: PropTypes.number,
    currentTotalResults: PropTypes.number,
    results: PropTypes.shape({}),
  }),
};
