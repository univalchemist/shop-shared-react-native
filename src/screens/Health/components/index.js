/* istanbul ignore file */

export {
  CallToActionLink,
  CardContainer,
  CardLoader,
  ErrorCard,
  TipCard,
} from './widgets';

export { default as ProductRecommendation } from './ProductRecommendation';
export { default as LifestyleTipsModal } from './LifestyleTipsModal';
export { default as LifestyleResults } from './LifestyleResults';
export { default as HealthScoreHistory } from './HealthScoreHistory';
export { default as HealthScore } from './HealthScore';
export { default as HealthResultCard } from './HealthResultCard';
export { default as GeneralTips } from './GeneralTips';
export { default as FaceAgingResults } from './FaceAgingResults';
