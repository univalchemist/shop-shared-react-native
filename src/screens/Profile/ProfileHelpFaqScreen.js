import {
  Box,
  Text,
  TrackedListItem,
  Image,
  ErrorPanel,
} from '@wrappers/components';
import React from 'react';
import { ScrollView } from 'react-native';
import { chevronRightArrow } from '@images';
import { PROFILE_HELP_FAQ_DETAILS } from '@routes';
import { categories } from '@store/analytics/trackingActions';

const ProfileHelpFaqScreen = ({ navigation, route }) => {
  const faqs = route?.params?.faqs;
  if (!faqs || !faqs.length) return <ErrorPanel />;

  return (
    <Box bg="gray.7" flex={1}>
      <ScrollView>
        <Box flex={1}>
          {faqs.map(faq => (
            <TrackedListItem
              key={faq.name}
              rightIcon={<Image source={chevronRightArrow} />}
              onPress={() =>
                navigation.navigate(PROFILE_HELP_FAQ_DETAILS, { faq })
              }
              accessible={true}
              accessibilityLabel={faq.name}
              actionParams={{
                category: categories.PROFILE_HELP,
                action: `Select FAQ question`,
                question_name: faq.name,
              }}
            >
              <Text>{faq.name}</Text>
            </TrackedListItem>
          ))}
        </Box>
      </ScrollView>
    </Box>
  );
};

export default ProfileHelpFaqScreen;
