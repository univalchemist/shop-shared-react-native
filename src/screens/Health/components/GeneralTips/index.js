import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { GeneralTipsCarousel } from './GeneralTipsCarousel';
import { GeneralTipsLoader } from './GeneralTipsLoader';
import { PromiseStatus } from '@middlewares';
import { Dimensions } from 'react-native';
import { useTheme } from '@wrappers/core/hooks';
import { ErrorCard } from '@screens/Health/components/widgets/ErrorCard';
import { Box, Text } from '@wrappers/components';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { FormattedMessage } from 'react-intl';
import { fetchLifestyleTips } from '@store/health/actions';

export const GeneralTips = ({ lifestyleTips, fetchLifestyleTips, locale }) => {
  const theme = useTheme();
  const horizontalMarginBetweenCards = theme.space[3];
  const viewportWidth = Dimensions.get('window').width;
  const viewportPadding = theme.space[4];
  const cardWidth = viewportWidth - viewportPadding * 2;
  const cardWidthWithSpacing = cardWidth + horizontalMarginBetweenCards;
  const cardHeight = 252.5;
  const viewportPaddingInsideCard = viewportPadding + 24;
  const loaderWidthInsideCard = viewportWidth - viewportPaddingInsideCard * 2;

  useEffect(() => {
    fetchLifestyleTips();
  }, [locale]);

  const renderGeneralTips = () => {
    switch (lifestyleTips.status) {
      case PromiseStatus.SUCCESS: {
        return lifestyleTips.tips.general ? (
          <GeneralTipsCarousel
            data={lifestyleTips.tips.general}
            cardWidthWithSpacing={cardWidthWithSpacing}
            cardHeight={cardHeight}
            horizontalMarginBetweenCards={horizontalMarginBetweenCards}
            cardWidth={cardWidth}
            viewportWidth={viewportWidth}
          />
        ) : (
          <Box
            width={cardWidthWithSpacing}
            height={cardHeight}
            px={horizontalMarginBetweenCards / 3}
            marginTop={16}
            alignSelf="center"
          >
            <ErrorCard color={'#FFF'} />
          </Box>
        );
      }
      case PromiseStatus.ERROR: {
        return (
          <Box
            width={cardWidthWithSpacing}
            height={cardHeight}
            px={horizontalMarginBetweenCards / 3}
            marginTop={16}
            alignSelf="center"
          >
            <ErrorCard color={'#FFF'} />
          </Box>
        );
      }
      default: {
        return (
          <GeneralTipsLoader
            cardWidthWithSpacing={cardWidthWithSpacing}
            cardHeight={cardHeight}
            horizontalMarginBetweenCards={horizontalMarginBetweenCards}
            cardWidth={cardWidth}
            viewportWidth={viewportWidth}
            loaderWidthInsideCard={loaderWidthInsideCard}
          />
        );
      }
    }
  };

  return (
    <>
      <Box>
        <Text
          color="gray.0"
          fontWeight={theme.fontWeights.bold}
          fontSize={16}
          lineHeight={22}
          letterSpacing={0.3}
          ml={32}
          mt={40}
        >
          <FormattedMessage id="health.sectionTitle.generalTips" />
        </Text>
      </Box>
      {renderGeneralTips()}
    </>
  );
};

const mapStateToProps = ({ health: { lifestyleTips }, intl: { locale } }) => ({
  lifestyleTips,
  locale,
});
const mapDispatchToProps = { fetchLifestyleTips };
const enhance = compose(connect(mapStateToProps, mapDispatchToProps));

GeneralTips.propTypes = {
  lifestyleTips: PropTypes.shape({
    tips: PropTypes.shape({ general: PropTypes.array }),
    status: PropTypes.string,
  }),
};

export default enhance(GeneralTips);
