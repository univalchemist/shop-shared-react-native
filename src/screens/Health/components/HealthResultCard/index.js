import React from 'react';
import PropTypes from 'prop-types';
import {
  healthIndicatorGreen,
  healthIndicatorPurple,
  healthIndicatorOrange,
  healthIndicatorRed,
  alcoholHighRisk,
  alcoholLowRisk,
  exerciseHighRisk,
  exerciseLowRisk,
  smokingLowRisk,
  smokingHighRisk,
  sleepLowRisk,
  sleepHighRisk,
  nutritionLowRisk,
  nutritionHighRisk,
  nutritionModerateRisk,
  mentalHealthLowRisk,
  mentalHealthHighRisk,
  mentalHealthModerateRisk,
} from '@images';
import { StyleSheet, View } from 'react-native';
import { Card } from 'react-native-elements';
import { Text, Image } from '@wrappers/components';
import theme from '@theme';
import { FormattedMessage } from 'react-intl';

const styles = StyleSheet.create({
  cardStyle: {
    width: 200,
    display: 'flex',
    borderRadius: 4,
    margin: 5,
    flex: 1,
    borderColor: 'white',
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.1,
    shadowRadius: 3.84,

    elevation: 5,
  },
  indicatorStyles: {
    justifyContent: 'center',
    display: 'flex',
    flexDirection: 'row',
    paddingTop: 8,
    alignContent: 'space-between',
  },
  labelStyle: {
    fontWeight: '600',
    paddingLeft: 7,
    letterSpacing: 1.6,
    fontSize: theme.fontSizes[0],
    lineHeight: 16,
  },
  healthStatusStyles: {
    paddingTop: 19,
    alignItems: 'center',
    display: 'flex',
    flexDirection: 'column',
  },
  healthContextualDataStyles: {
    paddingBottom: 12,
  },
  scoreStyles: {
    fontSize: theme.fontSizes[5],
    fontWeight: 'bold',
    lineHeight: theme.lineHeights[5],
    letterSpacing: -1.5,
    height: 40,
  },
  healthImageStyle: {
    height: 40,
  },
  contextualDataStyles: {
    fontSize: theme.fontSizes[1],
    color: theme.colors.gray[1],
  },
  classStyles: {
    fontSize: theme.fontSizes[1],
    fontWeight: 'bold',
    lineHeight: theme.lineHeights[4],
    letterSpacing: 0.05,
    textAlign: 'center',
    minHeight: 48,
  },
});

const healthImages = {
  smokingLowRisk: smokingLowRisk,
  smokingHighRisk: smokingHighRisk,
  alcoholLowRisk: alcoholLowRisk,
  alcoholHighRisk: alcoholHighRisk,
  exerciseLowRisk: exerciseLowRisk,
  exerciseHighRisk: exerciseHighRisk,
  sleepHighRisk: sleepHighRisk,
  sleepLowRisk: sleepLowRisk,
  nutritionHighRisk: nutritionHighRisk,
  nutritionLowRisk: nutritionLowRisk,
  nutritionModerateRisk: nutritionModerateRisk,
  mentalHealthHighRisk: mentalHealthHighRisk,
  mentalHealthLowRisk: mentalHealthLowRisk,
  mentalHealthModerateRisk: mentalHealthModerateRisk,
};

const healthIndicatorColor = {
  Healthy: healthIndicatorGreen,
  Overweight: healthIndicatorPurple,
  Obese: healthIndicatorOrange,
  ExtremelyObese: healthIndicatorRed,
  LowRisk: healthIndicatorGreen,
  HighRisk: healthIndicatorOrange,
  VeryHighRisk: healthIndicatorRed,
  smokingLowRisk: healthIndicatorGreen,
  smokingHighRisk: healthIndicatorRed,
  alcoholLowRisk: healthIndicatorGreen,
  alcoholHighRisk: healthIndicatorRed,
  exerciseLowRisk: healthIndicatorGreen,
  exerciseHighRisk: healthIndicatorRed,
  sleepLowRisk: healthIndicatorGreen,
  sleepHighRisk: healthIndicatorRed,
  nutritionHighRisk: healthIndicatorRed,
  nutritionLowRisk: healthIndicatorGreen,
  nutritionModerateRisk: healthIndicatorOrange,
  mentalHealthHighRisk: healthIndicatorRed,
  mentalHealthLowRisk: healthIndicatorGreen,
  mentalHealthModerateRisk: healthIndicatorOrange,
};

const HealthResultCard = ({
  healthLabel,
  healthScore,
  healthClass,
  healthContextualData,
}) => (
  <Card containerStyle={styles.cardStyle}>
    <View style={styles.indicatorStyles}>
      <Image
        source={healthIndicatorColor[healthClass]}
        resizeMode="contain"
        width={18}
        height={18}
      />
      <Text style={styles.labelStyle}>{healthLabel}</Text>
    </View>
    <View style={styles.healthStatusStyles}>
      {healthScore ? (
        <Text style={styles.scoreStyles}>{healthScore}</Text>
      ) : (
        <Image
          style={styles.healthImageStyle}
          source={healthImages[healthClass]}
          resizeMode="contain"
          height={18}
        />
      )}
      <Text style={styles.classStyles}>
        <FormattedMessage id={`health.class.${healthClass}`} />
      </Text>
    </View>
    <View style={styles.healthContextualDataStyles}>
      <Text style={styles.contextualDataStyles}>{healthContextualData}</Text>
    </View>
  </Card>
);

HealthResultCard.propTypes = {
  healthLabel: PropTypes.string.isRequired,
  healthScore: PropTypes.number,
  healthClass: PropTypes.string.isRequired,
  healthContextualData: PropTypes.string.isRequired,
};

export default HealthResultCard;
