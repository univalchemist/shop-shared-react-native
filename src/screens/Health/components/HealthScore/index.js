import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import HealthScoreError from './HealthScoreError';
import HealthScoreDisplay, { HealthScoreLoading } from './HealthScoreDisplay';
import { PromiseStatus } from '@middlewares';
import { Card } from '@wrappers/components';
import { getUserHealthScore } from '@store/health/actions';

export const HealthScore = ({ healthScore = {}, getUserHealthScore }) => {
  useEffect(() => {
    const fetchUserHealthScore = async () => {
      await getUserHealthScore();
    };
    fetchUserHealthScore();
  }, [getUserHealthScore]);

  const { score, status } = healthScore;
  const isValidScore = !isNaN(parseInt(score, 10));
  const showHealthScore = () => {
    if (status === PromiseStatus.START) {
      return <HealthScoreLoading />;
    }
    if (status === PromiseStatus.ERROR || !isValidScore) {
      return <HealthScoreError />;
    }
    if (status === PromiseStatus.SUCCESS) {
      return <HealthScoreDisplay score={score} />;
    }
    return null;
  };

  return (
    <Card bg="white" flexGrow={1} px={4} pt={4} pb={24}>
      {showHealthScore()}
    </Card>
  );
};

const mapStateToProps = ({ health: { healthScore } }) => ({
  healthScore,
});

export default connect(mapStateToProps, { getUserHealthScore })(HealthScore);
