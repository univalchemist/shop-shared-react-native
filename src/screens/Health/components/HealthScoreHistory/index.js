import React, { useEffect } from 'react';
import HealthScoreHistoryGraph, {
  HealthScoreHistoryLoading,
} from './HealthScoreHistoryGraph';
import { connect } from 'react-redux';
import { PromiseStatus } from '@middlewares';
import { getUserHealthScoreHistory } from '@store/health/actions';
import HealthScoreHistoryError from './HealthScoreHistoryError';
import { Card } from '@wrappers/components';

export const HealthScoreHistory = ({
  healthScoreHistory = {},
  getUserHealthScoreHistory,
}) => {
  useEffect(() => {
    const fetchUserHealthScoreHistory = async () => {
      await getUserHealthScoreHistory();
    };
    fetchUserHealthScoreHistory();
  }, [getUserHealthScoreHistory]);

  const { data, status } = healthScoreHistory;
  const showHealthScoreHistory = () => {
    if (status === PromiseStatus.START) {
      return <HealthScoreHistoryLoading />;
    }
    if (status === PromiseStatus.ERROR) {
      return <HealthScoreHistoryError />;
    }
    if (status === PromiseStatus.SUCCESS) {
      return <HealthScoreHistoryGraph data={data} />;
    }
    return null;
  };

  return (
    <Card bg="white" flexGrow={1} justifyContent="center" px={3} pt={3} pb={3}>
      {showHealthScoreHistory()}
    </Card>
  );
};

const mapStateToProps = ({ health: { healthScoreHistory } }) => ({
  healthScoreHistory,
});

export default connect(mapStateToProps, { getUserHealthScoreHistory })(
  HealthScoreHistory,
);
