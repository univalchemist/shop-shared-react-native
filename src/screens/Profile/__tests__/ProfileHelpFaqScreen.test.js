import React from 'react';
import { renderForTestWithStore } from '@testUtils';
import { flushMicrotasksQueue, fireEvent } from 'react-native-testing-library';
import ProfileHelpFaqScreen from '../ProfileHelpFaqScreen';
import { PROFILE_HELP_FAQ_DETAILS } from '@routes';
import { ErrorPanel } from '@wrappers/components';

const faqs = [
  {
    name: 'FaqSection1',
    content: 'FaqContent1',
  },
  {
    name: 'FaqSection2',
    content: 'FaqContent2',
  },
];

const navigation = { navigate: jest.fn() };
const route = { params: { faqs } };

describe('ProfileHelpFaqScreen', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should display correctly when data successfully loads', async () => {
    const [Component] = renderForTestWithStore(
      <ProfileHelpFaqScreen navigation={navigation} route={route} />,
      {},
    );

    await flushMicrotasksQueue();

    expect(Component.getByText('FaqSection1')).toBeDefined();
    expect(Component.getByText('FaqSection2')).toBeDefined();
    expect(Component.toJSON()).toMatchSnapshot();
  });

  it('should navigate to the FaqDetails screen', async () => {
    const [Component] = renderForTestWithStore(
      <ProfileHelpFaqScreen navigation={navigation} route={route} />,
      {},
    );
    await flushMicrotasksQueue();
    const tappableFaq = Component.getByText('FaqSection1');
    await fireEvent.press(tappableFaq);

    expect(navigation.navigate).toBeCalledTimes(1);
    expect(navigation.navigate).toBeCalledWith(PROFILE_HELP_FAQ_DETAILS, {
      faq: faqs[0],
    });
  });

  describe('Error handling', () => {
    const faqsStates = [
      ['faqs is undefined', undefined],
      ['faqs is null', null],
      ['faqs is empty', {}],
    ];

    test.each(faqsStates)(
      'should display the error panel when %s',
      async (_, faqState) => {
        const newRoute = {
          ...route,
          params: {
            faqs: faqState,
          },
        };
        const [Component] = renderForTestWithStore(
          <ProfileHelpFaqScreen navigation={navigation} route={newRoute} />,
          {},
        );

        await flushMicrotasksQueue();

        const errorPanel = Component.getByType(ErrorPanel);
        expect(errorPanel).toBeDefined();
      },
    );
  });

  it('should render error panel when faqs is null', async () => {
    const newRoute = {
      ...route,
      params: {
        faqs: null,
      },
    };
    const [Component] = renderForTestWithStore(
      <ProfileHelpFaqScreen navigation={navigation} route={newRoute} />,
      {},
    );
    await flushMicrotasksQueue();
    const errorPanel = Component.getByType(ErrorPanel);
    expect(errorPanel).toBeDefined();
  });

  it('should render error panel when faqs is empty', async () => {
    const newRoute = {
      ...route,
      params: {
        faqs: [],
      },
    };
    const [Component] = renderForTestWithStore(
      <ProfileHelpFaqScreen navigation={navigation} route={newRoute} />,
      {},
    );
    await flushMicrotasksQueue();
    const errorPanel = Component.getByType(ErrorPanel);
    expect(errorPanel).toBeDefined();
  });
});
