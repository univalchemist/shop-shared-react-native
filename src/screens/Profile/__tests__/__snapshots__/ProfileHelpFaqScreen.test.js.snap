// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileHelpFaqScreen should display correctly when data successfully loads 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              enabled={false}
              hitSlop={
                Object {
                  "right": -700,
                }
              }
              maxDeltaY={20}
              minOffsetX={5}
              onHandlerStateChange={[Function]}
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": -1,
                      },
                      "shadowOpacity": 0.3,
                      "shadowRadius": 5,
                      "top": 0,
                      "width": 3,
                    }
                  }
                />
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        bg="gray.7"
                        flex={1}
                        style={
                          Array [
                            Object {
                              "backgroundColor": "#f9f9f9",
                              "flexBasis": 0,
                              "flexGrow": 1,
                              "flexShrink": 1,
                            },
                          ]
                        }
                      >
                        <RCTScrollView>
                          <View>
                            <View
                              flex={1}
                              style={
                                Array [
                                  Object {
                                    "flexBasis": 0,
                                    "flexGrow": 1,
                                    "flexShrink": 1,
                                  },
                                ]
                              }
                            >
                              <View
                                accessibilityLabel="FaqSection1"
                                accessible={true}
                                focusable={true}
                                isTVSelectable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={null}
                              >
                                <View
                                  alignItems="center"
                                  flexDirection="row"
                                  px={4}
                                  py={24}
                                  style={
                                    Array [
                                      Object {
                                        "alignItems": "center",
                                        "flexDirection": "row",
                                        "paddingBottom": 24,
                                        "paddingLeft": 32,
                                        "paddingRight": 32,
                                        "paddingTop": 24,
                                      },
                                    ]
                                  }
                                >
                                  <View
                                    flex={1}
                                    style={
                                      Array [
                                        Object {
                                          "flexBasis": 0,
                                          "flexGrow": 1,
                                          "flexShrink": 1,
                                        },
                                      ]
                                    }
                                  >
                                    <Text
                                      color="gray.1"
                                      fontFamily="default"
                                      fontSize={16}
                                      lineHeight={22}
                                      style={
                                        Array [
                                          Object {
                                            "color": "#646464",
                                            "fontFamily": "Arial",
                                            "fontSize": 16,
                                            "lineHeight": 22,
                                          },
                                        ]
                                      }
                                    >
                                      FaqSection1
                                    </Text>
                                  </View>
                                  <View
                                    marginLeft={3}
                                    style={
                                      Array [
                                        Object {
                                          "marginLeft": 16,
                                        },
                                      ]
                                    }
                                  >
                                    <Image
                                      source={
                                        Object {
                                          "testUri": "../../../src/images/chevronRightArrow.png",
                                        }
                                      }
                                      style={
                                        Array [
                                          Object {},
                                        ]
                                      }
                                    />
                                  </View>
                                </View>
                                <View
                                  pl={4}
                                  pr={4}
                                  style={
                                    Array [
                                      Object {
                                        "paddingLeft": 32,
                                        "paddingRight": 32,
                                      },
                                    ]
                                  }
                                >
                                  <View
                                    backgroundColor="gray.5"
                                    height={1}
                                    style={
                                      Array [
                                        Object {
                                          "backgroundColor": "#e3e3e3",
                                          "height": 1,
                                          "width": "100%",
                                        },
                                      ]
                                    }
                                    width="100%"
                                  />
                                </View>
                              </View>
                              <View
                                accessibilityLabel="FaqSection2"
                                accessible={true}
                                focusable={true}
                                isTVSelectable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={null}
                              >
                                <View
                                  alignItems="center"
                                  flexDirection="row"
                                  px={4}
                                  py={24}
                                  style={
                                    Array [
                                      Object {
                                        "alignItems": "center",
                                        "flexDirection": "row",
                                        "paddingBottom": 24,
                                        "paddingLeft": 32,
                                        "paddingRight": 32,
                                        "paddingTop": 24,
                                      },
                                    ]
                                  }
                                >
                                  <View
                                    flex={1}
                                    style={
                                      Array [
                                        Object {
                                          "flexBasis": 0,
                                          "flexGrow": 1,
                                          "flexShrink": 1,
                                        },
                                      ]
                                    }
                                  >
                                    <Text
                                      color="gray.1"
                                      fontFamily="default"
                                      fontSize={16}
                                      lineHeight={22}
                                      style={
                                        Array [
                                          Object {
                                            "color": "#646464",
                                            "fontFamily": "Arial",
                                            "fontSize": 16,
                                            "lineHeight": 22,
                                          },
                                        ]
                                      }
                                    >
                                      FaqSection2
                                    </Text>
                                  </View>
                                  <View
                                    marginLeft={3}
                                    style={
                                      Array [
                                        Object {
                                          "marginLeft": 16,
                                        },
                                      ]
                                    }
                                  >
                                    <Image
                                      source={
                                        Object {
                                          "testUri": "../../../src/images/chevronRightArrow.png",
                                        }
                                      }
                                      style={
                                        Array [
                                          Object {},
                                        ]
                                      }
                                    />
                                  </View>
                                </View>
                                <View
                                  pl={4}
                                  pr={4}
                                  style={
                                    Array [
                                      Object {
                                        "paddingLeft": 32,
                                        "paddingRight": 32,
                                      },
                                    ]
                                  }
                                >
                                  <View
                                    backgroundColor="gray.5"
                                    height={1}
                                    style={
                                      Array [
                                        Object {
                                          "backgroundColor": "#e3e3e3",
                                          "height": 1,
                                          "width": "100%",
                                        },
                                      ]
                                    }
                                    width="100%"
                                  />
                                </View>
                              </View>
                            </View>
                          </View>
                        </RCTScrollView>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
